chia/seeder/dns_server.py

```diff
@@ -395,37 +404,72 @@ async def stop(self) -> None:
     async def periodically_get_reliable_peers(self) -> None:
         sleep_interval = 0
         while not self.shutdown_event.is_set() and self.crawl_store is not None:
+            await self.refresh_reliable_peers()
+            sleep_interval = min(15, sleep_interval + 1)
+            await asyncio.sleep(sleep_interval * 60)
+
+    async def refresh_reliable_peers(self) -> None:
+        if self.crawl_store is None:
+            return
+        new_reliable_peers: list[str] = []
+        while True:
             try:
                 new_reliable_peers = await self.crawl_store.get_good_peers()
             except Exception as e:
                 log.error(f"Error loading reliable peers from database: {e}. Traceback: {traceback.format_exc()}.")
+                await asyncio.sleep(2)
```

Comment:
if we get here. we may get stuck indefinitely, not shutting down even when `sef.shutdown_event.is_set()`